openfe/setup/atommapping.py

```diff
@@ -1,8 +1,41 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+from rdkit import Chem
+from typing import Dict, Union, TypeVar
+
+RDKitMol = TypeVar('RDKitMol')
+
+
 class AtomMapping:
-    @classmethod
-    def from_perses(cls, perses_mapping):
-        pass
+    """Simple container with the mapping between two Molecules
+
+    Attributes
+    ----------
+    molA, molB : rdkit.Mol
+      the two Molecules in the mapping
+    AtoB, BtoA : dict
+      maps the index of an atom in either molecule **A** or **B** to the other.
+      If this atom has no corresponding atom, None is returned.
 
-    def __getitem__(self, other):
-        return None
 
+    The size of molecule A/B is given by the length of the AtoB/BtoA dictionary
+    """
+    molA: RDKitMol
+    molB: RDKitMol
+    AtoB: Dict[int, Union[int, None]]
+    BtoA: Dict[int, Union[int, None]]
+
+    def __init__(self, molA: RDKitMol, molB: RDKitMol):
```

Comment:
Any reason not to include the dicts in the init sig? I think it's easier for a user to write:

```python
return AtomMapping(molA, molB, myAtoB, myBtoA)
```

than

```python
mapping = AtomMapping(molA, molB)
mapping.AtoB = myAtoB
mapping.BtoA = myBtoA
return mapping
```

If you're worried about users mixing up order, make them keyword required. If not, maybe the whole thing is `@dataclass`?